crates/polars-plan/src/plans/aexpr/scalar.rs

```diff
@@ -8,10 +8,14 @@ pub fn is_scalar_ae(node: Node, expr_arena: &Arena<AExpr>) -> bool {
         AExpr::Literal(lv) => lv.is_scalar(),
         AExpr::Function { options, input, .. }
         | AExpr::AnonymousFunction { options, input, .. } => {
-            if options.is_elementwise() || !options.flags.contains(FunctionFlags::CHANGES_LENGTH) {
+            if options.flags.contains(FunctionFlags::RETURNS_SCALAR) {
+                true
+            } else if options.is_elementwise()
+                || !options.flags.contains(FunctionFlags::CHANGES_LENGTH)
+            {
                 input.iter().all(|e| e.is_scalar(expr_arena))
             } else {
-                options.flags.contains(FunctionFlags::RETURNS_SCALAR)
+                false
             }
```

Comment:
re-ordered this condition to fix an issue where `pl.col("x").null_count() > 0` was not being identified as scalar